packages/css/src/plugins/at-rules/parsers.ts

```diff
@@ -0,0 +1,231 @@
+    case 'max-device-width':
+      // max-width: XYZ
+      // is the same as
+      // width <= XYZ
+      return {
+        property: property === 'max-width' ? 'width' : 'device-width',
+        comparisonOperator: '<=',
+      };
+    case 'min-height':
+    case 'min-device-height':
+      return {
+        property: property === 'min-height' ? 'height' : 'device-height',
+        comparisonOperator: '>=',
+      };
+    case 'max-height':
+    case 'max-device-height':
+      return {
+        property: property === 'max-height' ? 'height' : 'device-height',
+        comparisonOperator: '<=',
+      };
+    case undefined:
+      return;
+    default:
+      throw new SyntaxError(`Unexpected property '${property}' found when sorting media queries`);
+  }
+};
+
+/**
+ * This runs when we expect
+ *
+ *     <width|height> <comparisonOperator> <length>
+ *
+ * but we actually got
+ *
+ *     <length> <comparisonOperator> <width|height>
+ *
+ * So we need to normalise the comparison operator by reversing it.
+ *
+ * @param operator Comparison operator.
+ * @returns The reversed comparison operator.
+ */
+const reverseComparisonOperator = (operator: ComparisonOperator): ComparisonOperator => {
+  switch (operator) {
+    case '<':
+      return '>';
+    case '>':
+      return '<';
+    case '<=':
+      return '>=';
+    case '>=':
+      return '<=';
+    case '=':
+      return '=';
+  }
+};
+
+const getProperty = (match: RegExpMatchArray): PropertyInfo | undefined => {
+  const property = match.groups?.property;
+
+  if (property === undefined) {
+    return;
+  }
+
+  if (isPropertyValid(property)) {
+    return {
+      property,
+    };
+  }
+
+  throw new SyntaxError(`Unexpected property '${property}' found when sorting media queries.`);
+};
+
+const getOperator = (
+  match: RegExpMatchArray,
+  reverse: 'reverse' | 'no-reverse',
+  groupName = 'operator'
+): OperatorInfo | undefined => {
+  const operator = match.groups?.[groupName];
+
+  if (operator === undefined) {
+    return;
+  }
+
+  if (isComparisonOperatorValid(operator)) {
+    return {
+      comparisonOperator: reverse === 'reverse' ? reverseComparisonOperator(operator) : operator,
+    };
+  }
+
+  throw new SyntaxError(
+    `Unexpected comparison operator '${operator}' found when sorting media queries.`
+  );
+};
+
+const getLengthInfo = (match: RegExpMatchArray, groupName = 'length'): LengthInfo | undefined => {
+  const length_ = match.groups?.[groupName];
+  const lengthUnit = match.groups?.[groupName];
+
+  if (typeof length_ !== 'string' || typeof lengthUnit !== 'string') {
+    return undefined;
+  }
+  return {
+    length: +length_,
+    lengthUnit: lengthUnit as LengthUnit,
+  };
+};
+
+const getBasicMatchInfo = (match: RegExpMatchArray): BasicMatchInfo | undefined => {
+  if (!match.index) {
+    return;
+  }
+
+  return { index: match.index, match: match[0] };
+};
+
+export const parseSituationOne = (match: RegExpMatchArray): ParsedAtRule | undefined => {
```

Comment:
Having these names as what they're working with would be more ideal. None of us will remember what each situation is later on (yes it's somewhere above in a comment but still).